crates/red_knot_python_semantic/src/types/builder.rs

```diff
@@ -326,7 +328,7 @@ mod tests {
         let t1 = Type::IntLiteral(1);
         let union = UnionType::from_elements(&db, [t0, t1]).expect_union();
 
-        assert_eq!(union.elements_vec(&db), &[t0, t1]);
+        assert_eq!(union.elements(&db).as_ref(), &[t0, t1]);
```

Comment:
you could also do this with

```suggestion
        assert_eq!(**union.elements(&db), [t0, t1]);
```

(etc. for all the other `.as_ref()`s you added for the `assert_eq!` calls in this file)

Reply:
I think I'll go with Micha's suggestion of a slice-returning accessor to eliminate the need for either of these.